include/seqan3/alphabet/quality/quality_base.hpp

```diff
@@ -0,0 +1,195 @@
+// ============================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ============================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Marie Hoffmann <marie.hoffmann AT fu-berlin.de>
+ * \brief Contains seqan3::phred42 quality scores.
+ */
+
+#pragma once
+
+
+#include <seqan3/alphabet/detail/alphabet_base.hpp>
+#include <seqan3/alphabet/detail/convert.hpp>
+#include <seqan3/alphabet/quality/concept.hpp>
+
+namespace seqan3
+{
+
+/*!\brief A CRTP-base that refines seqan3::alphabet_base and is used by the quality alphabets.
+ * \tparam derived_type The CRTP parameter type.
+ * \tparam size         The size of the alphabet.
+ * \tparam char_t       The character type of the alphabet (set this to `void` when defining just a
+ *                      seqan3::semi_alphabet_concept).
+ */
+template <typename derived_type, auto size, typename char_t = char>
+class quality_base : public alphabet_base<derived_type, size, char_t>
```

Comment:
All our quality stuff is phred based, there is no difference. It's not usable by `qualified` because `qualified` is a cartesian composition. This is the same as `nucleotide_base` which is also not used by all nucleotide composition, just the "core alphabets".

Reply:
Thank you for clarifying! I would still propose the name change, because I changed the test name in #523 from `quality_test_template` to `phred_test_template`, because it was confusing that a test called `quality` in a subfolder named `quality` wasn't used `qualified` tests. I think you even remarked this in #535.